src/commands/createFunction/openAPISteps/OpenAPICreateStep.ts

```diff
@@ -0,0 +1,116 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { OpenDialogOptions, ProgressLocation, Uri, window, workspace } from "vscode";
+import { AzureWizardExecuteStep } from "vscode-azureextensionui";
+import { ProjectLanguage } from "../../../constants";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../localize";
+import { cpUtils } from "../../../utils/cpUtils";
+import { IFunctionWizardContext } from "../IFunctionWizardContext";
+
+export class OpenAPICreateStep extends AzureWizardExecuteStep<IFunctionWizardContext> {
+    public priority: number = 220;
+
+    public async execute(wizardContext: IFunctionWizardContext): Promise<void> {
+        const uris: Uri[] = await this.askDocument();
+        const uri: Uri = uris[0];
+        const args: string[] = [];
+
+        const errorMessage: string = 'Please make sure you have Autorest installed. Visit https://aka.ms/autorest to find more information on Autorest and installation steps.';
+
+        if (!await isNPMInstalled()) {
+            throw new Error(localize("notSupported", `NPM is not installed. ${errorMessage}`));
+        } else if (!await isAutoRestInstalled()) {
+            throw new Error(localize("notSupported", `Autorest could not be detected on the system. ${errorMessage}`));
+        }
+
+        let namespace: string = '';
+        if (wizardContext.language === ProjectLanguage.CSharp || wizardContext.language === ProjectLanguage.Java) {
```

Comment:
All prompts should be a part of AzureWizardPromptSteps, not AzureWizardExecuteSteps (for various reasons, including the step count and back button).  For `askDocument`, you'll have to create your own prompt step. For `askNamespace`, you should be able to use `DotnetNamespaceStep` for C# and  `JavaPackageNameStep` for Java. Those steps should be added inside `FunctionListStep.getSubWizard` around [here](https://github.com/microsoft/vscode-azurefunctions/pull/2395/files#diff-66c8dd99a66b6c46145ef571643118e6R121), similar to [here](https://github.com/microsoft/vscode-azurefunctions/blob/main/src/commands/createFunction/FunctionListStep.ts#L79)

Reply:
Thanks a lot for the pointers - I was able to use them and simplify my code a lot. :)